File: Problems_and_Solutions/day-7/Java/Solution3.java
class Solution3 {

    int front = 0, rear = 0;
    int arr[] = new int[100005];

    void push(int x) {
        if (rear == arr.length - 1) return;
        arr[rear++] = x;
    }

    int pop() {
        if (front == rear) return -1;
        return arr[front++];
    }

    public static void main(String[] args) {
        Solution3 q = new Solution3();

        q.push(10);
        q.push(20);
        q.push(30);

        System.out.println(q.pop());
        System.out.println(q.pop());
        System.out.println(q.pop());
        System.out.println(q.pop());
    }
}

File: Problems_and_Solutions/day-7/cpp/solution 3.cpp
void MyQueue ::push(int x) {
    // Your Code
    
        arr[rear]=x;
        rear=(rear+1)%100005;
}

// Function to pop an element from queue and return that element.
int MyQueue ::pop() {
    // Your Code
    int res=front==rear?-1:arr[front];
    if(res!=-1)
        front=(front+1)%100005;
    return res;
}
